### PR TITLE
IPsec: Generate strongswan.conf Radius backend configuration from mobile client page

### PR DESCRIPTION
**Pull request by stumbaumr**
Hi,

this is the PR for https://github.com/opnsense/core/issues/3339 .

If there is a mobile phase 1 containing a EAP-Radius configuration the system behaves as before.

If there is no mobile phase 1 but a Radius server configured in the VPN-IPsec-Mobile client page the radius plugin config will be configured from there.

If the plan is to have - at some point in the future - more than only one mobile phase this is the first step  (remove the configuration of the per-user based authentication backends away from the tunnel config into the Mobile client config).

### CODE REVIEW

**Comment by AdSchellevis** [comment]
I'm not sure we should do this now, the code underneath already doesn't win a beauty contest adding more logic without functional use probably doesn't make it better.

Supporting more mobile phase1 connections (using different authentication schemes) in the future would certainly be a nice goal, although likely very difficult to reach with how things work now. 
If possible, we should start with figuring out how the different currently supported authentication options should function when there's more than one (for the old xauth this was not possible as far as I know, with pam in play there might be more options).

It's quite a project to restructure the current code base for ipsec to something more maintainable, not sure when we're going to do so.
There are some ideas, but not concrete yet. Questions like this popup from time to time (https://github.com/opnsense/core/issues/2833), but investments needed to make it happen are quite substantial.

**Comment by stumbaumr** [comment]
So what is your usual approach when doing bigger overhauls like this? Would you accept a PR which fixes everything in one go? As far as I understand @mimugmail is based in Munich, too. If we organise a hackfest that replaces the whole non-PSK/RSA authentications - would the outcome of that be acceptable?

Or do you prefer smaller code changes in a step-by-step fashion? Like the above?

**Comment by mimugmail** [comment]
Please note that I'm not a core member and will never be, as I'm already in partner state.

**Comment by AdSchellevis** [comment]
I might do some work later on this topic, wrap logic in easier to understand classes like we did for the firewall code. Small commits functionally adding something are still being considered. 

Partners help fund the project, which we also take into consideration when prioritising work. Thanks @mimugmail !

**Comment by stumbaumr** [comment]
So what I understand here is "lean back, wait for me to fix it". Fine.

**Comment by AdSchellevis** [comment]
ok, let me rephrase, we do accept functional changes to the current code if it fixes actual issues or adds relevant features. 
We just don't accept partial refactoring pr's as eventually we probably will redo the whole module since it's difficult to maintain as it is and needs a solid plan before doing so. 

Things that help getting there in this case are investigating how current authentication scenarios would fit in proposed new ones (how do they look now, what should be changed, would that be possible within configuration parameters that are available now, etc, etc). Plan->Do->Check->Act

At the end of the day, all kinds of setups are supported with our system, we would like to avoid breakage unless absolutely necessary.
